### DISCUSSION TITLE
Creating and persisting an entity triggers updates on relationships

### DISCUSSION THREAD

**Discussion by julestruong** [mikro-orm/mikro-orm#4002] [Q&A] [answered]
Hi guys i have an odd behavior that's happening and i would like to know if this is normal behavior or a bug.

I have a `booking` entity which has relationships with `user` entity and `spot` entity.

```typescript
import { EventStatus } from './../../common/enum/status.enum';
import { Entity, Enum, ManyToOne, Property } from '@mikro-orm/core';
import BaseEntity from '../../../core/entity/base.entity';
import { MembershipEntity } from '../../../membership/entity/membership.entity';
import { SpotEntity } from '../../../place-folder/spot/entity/spot.entity';
import { UserEntity } from '../../../user/entity/user.entity';

@Entity({ tableName: 'booking' })
export class BookingEntity extends BaseEntity {
  // Properties
  @Property()
  start: Date;

  @Property()
  end: Date;

  // Relationships

  @ManyToOne(() => SpotEntity, {
    inversedBy: 'bookings',
    joinColumn: 'spot_id',
  })
  spot!: SpotEntity;

  @ManyToOne(() => UserEntity, {
    inversedBy: 'bookings',
    joinColumn: 'user_id',
  })
  user!: UserEntity;


  constructor(partial: Partial<BookingEntity>) {
    super();
    Object.assign(this, partial);
  }
}

```

I have this service that i call 

```typescript
constructor(
    @InjectRepository(BookingEntity)
    private bookingEntityRepository: EntityRepository<BookingEntity>,
  ) {}

  async create(data: NewBooking): Promise<Booking> {
    const bookingCreateParams: RequiredEntityData<BookingEntity> = {
      start: data.start,
      end: data.end,    
      user: data.user.id,
      spot: data.spot.id,
      createdAt: new Date(),
      updatedAt: new Date(),
    };

    const bookingEntity =
      this.bookingEntityRepository.create(bookingCreateParams);

    return await this.bookingEntityRepository.persistAndFlush(bookingEntity);
  }
```

I dunno why but this triggers 3 sql queries   
```sql
INSERT INTO `booking` VALUES(...)
UPDATE `user` id=3  WHERE id = 3
UPDATE `spot` id=4 WHERE id = 4
#those are not the real sql queries, this is just for you to understand that there are 2 more queries
```

Obviously the last two are useless. 
Is it something wrong i'm doing in some configuration ? 
Is there any ways of avoiding this behavior ?

**Comment by B4nan** [comment, marked as answer]
> Object.assign(this, partial);

Doing this is generally not a good idea, this will work only for scalar properties, you can't assign any relation property this way. Not sure if that's your problem, depends on how you use it.

Ideally, provide complete repro. Are you on latest version?

**Reply by julestruong** [reply]
noted for the `Object.assign` 
but removing it doesn't change things.

I'm using 5.6.0.

I'll try to make a complete repro.

**Reply by B4nan** [reply]
then upgrade first, there was something like this recently fixed, you never know... 8 patch releases are quite a lot of bug fixes you are missing out :]

**Reply by julestruong** [reply]
true, it seems there is a bugfix about extra updates in [5.6.7](https://github.com/mikro-orm/mikro-orm/releases/tag/v5.6.7) i'll try that

**Reply by julestruong** [reply]
Thank you ! It was a bug ;) 
And the upgrade fixed it.

**Comment by julestruong** [comment]
Hi @B4nan , is it me or is this thing got back with the 5.7.10 ?

Not in the same way tho, i'm just trying to make a `findOne` and somehow, there are `UPDATE` queries run .

**Reply by julestruong** [reply]
It looks more like this https://github.com/mikro-orm/mikro-orm/issues/3941. 
But i'm using the latest version and it still happens.

**Reply by B4nan** [reply]
It's a bit irrelevant what issue is similar, you will need to provide a reproduction either way, all bugs are closed with a test case, and all tests are passing...
